Add CI coverage for Windows

utpilla opened the pull request:
## Changes
- Update CI workflow to run jobs on Windows as well

Note: Certain jobs from the CI workflow can only be run on Linx for now

cijothomas commented on .github/workflows/ci.yml:
```diff
@@ -32,9 +34,12 @@ jobs:
-    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        os: [ windows-latest, ubuntu-latest ]
```
does it matter if lint checks are done on windows/linux/either?

utpilla replied:
Good point! Updated the workflow to run `docs` and `lint` jobs only on Linux.

lalitb reviewed (approved):
Thanks.

cijothomas commented:
Consider doing windows for only the test/stable for now, given it is much slower. We can add it for all, later.
Also see what other repos like tracing-rs is doing.

ThomsonTan commented on .github/workflows/ci.yml:
```diff
@@ -32,7 +34,7 @@ jobs:
     - uses: arduino/setup-protoc@v3
     - name: Test
-      run: ./scripts/test.sh
+      run: bash ./scripts/test.sh
```
Is there some scenario where the `bash` prefix is required?

utpilla replied:
`bash` prefix lets us run `.sh` files on Windows.

TommyCpp reviewed (approved):
Looks good. Thanks

utpilla commented:
> Consider doing windows for only the test/stable for now, given it is much slower. We can add it for all, later. Also see what other repos like tracing-rs is doing.

@cijothomas I checked the tracing repo and for non-Linux platforms, they only test using the stable version of Rust. Check [this](https://github.com/tokio-rs/tracing/blob/master/.github/workflows/CI.yml#L194-L210). I have updated the workflow to do the same. I have also updated code coverage workflow to only run on Linux.
